### DISCUSSION TITLE
Skip useQuery depending on result of mutation.mutate()

### DISCUSSION THREAD

**Discussion by bannndi** [TanStack/query#6107] [Q&A]
Hello guys!

Is there effective way (or even possible) to skip **query** request only for one or several mutation actions ?

`
Parent
  **query** = useQuery(requestFn)

  // Parent has shared state/logic with child
  
  Child
    Button onClick={post**(or put/patch)**.mutate()} // When clicked on button I want **query** in Parent to be skipped, because post response already has needed json data. All other times I want **query** to work as usual.
  Child
Parent
`

Had some experiments with queryClient.setQueryData(), MutationCache.state, query.enabled.

**Comment by Lisba** [comment]
Hi, @bannndi! You're looking for a direct update through [queryClient.setQueryData()](https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientsetquerydata).

There are two main approaches to update the cache after a mutation (post, put, patch, delete...):
1. The most known is through query invalidations where you indicate to tanstack query that some queries need to be refetched because of backend changes.
2. The other one is through direct updates where you already received the new data from the backend response and just want to update the cache directly avoiding those extra queries (because you already have the data, of course).

So instead of using a query invalidation, you can apply this direct update like this:

```
const mutation = useMutation({
    mutationFn: addNewProductFn,
    onSuccess: (newProduct) => {
      queryClient.setQueryData(['product', newProduct.id], newProduct)
    }
})
```

I hope this helps you.
